logstash-core/lib/logstash/modules_cli_parser.rb

```diff
@@ -0,0 +1,89 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/logging"
+require "logstash/errors"
+
+class LogStash::ModulesCLIParser
+  include LogStash::Util::Loggable
+
+  def initialize(mods, modvars)
+    @result = []
+    fixedvars = modvars.nil? || modvars == [nil] ? [] : modvars
+    self.parse_it(mods, fixedvars)
+  end
+
+  def output
+    @result
+  end
+
+  def decommafy(value)
+    retval = []
+    if value.split(',').length == 1
+      retval << value
+    elsif value.split(',').length > 1
+      value.split(',').each {|subvalue| retval << subvalue}
+    end
+    retval
+  end
```

Comment:
You're right.  Breaking this down into a class removed its need altogether.  I'll be resolving this with `#concat` and a single `split(',')` instead.